### PR TITLE
Added option to hide base length units in sketches.

### PR DESCRIPTION
**Pull request by hokieengr**
As discussed in [this forum](https://forum.freecadweb.org/viewtopic.php?f=8&t=29526&start=10).

Thank you!

### CODE REVIEW

File: src/Mod/Sketcher/Gui/ViewProviderSketch.cpp [hunk 1]
```diff
@@ -2745,6 +2747,74 @@ bool ViewProviderSketch::doubleClicked(void)
...
+    bool                            iHideUnits;
+    QString                         userStr; // final return string
+    QString                         unitStr;  // the actual unit string
+    QString                         baseUnitStr; // the expected base unit string
+    double                          factor; // unit scaling factor, currently not used
+    Base::UnitSystem                unitSys; // current unit system
+    
+    // Get value of HideUnits option. Default is false.
+    hGrpSketcher = App::GetApplication().GetUserParameter().GetGroup("BaseApp")->GetGroup("Preferences")->GetGroup("Mod/Sketcher");
+    iHideUnits = hGrpSketcher->GetBool("HideUnits", 0l);
```
Comment (yorikvanhavre) [inline src/Mod/Sketcher/Gui/ViewProviderSketch.cpp]: This sets the default to true, although in the ui the default is false (unchecked) Should be the same everywhere. I'd recommend the default to be false, so existing users don't experience an unwanted change

Comment (hokieengr) [reply src/Mod/Sketcher/Gui/ViewProviderSketch.cpp]: Good catch. Typo on my part: comments state what I meant to do as well as the GUI.

I'll update it when I get home unless it's an easy thing for you to fix (not sure what it looks like from your side).

**Comment by hokieengr** [comment]
@yorikvanhavre Should I be clicking "Update branch" whenever I notice it's out of date again or does that just cause more work for you?

**Comment by yorikvanhavre** [comment]
Hi, no need, now github offers us a one-click "rebase & merge" option which turns this easy to do on our side...

**Comment by yorikvanhavre** [comment]
Waiting for the build tests to complete and I'll merge this.

**Comment by hokieengr** [comment]
Sounds very fancy!

**Comment by yorikvanhavre** [comment]
merged

**Pull request closed without merging**